core/src/main/java/hudson/PluginManager.java

```diff
@@ -946,10 +895,20 @@ public synchronized void resolveDependantPlugins() {
                 for (Dependency dependency : dependencies) {
                     if (dependency.shortName.equals(plugin.getShortName())) {
                         dependants.add(possibleDependant.getShortName());
+
+                        // If, in addition, the dependency is optional, add to the optionalDependants list
+                        if (dependency.optional) {
+                            optionalDependants.add(possibleDependant.getShortName());
+                        }
+
+                        // already know possibleDependant depends on plugin, no need to continue with the rest of
+                        // dependencies. We continue with the next possibleDependant
+                        break;
```

Comment:
Please avoid using `break` in general (except in `switch`) as a good practice, to avoid attracting C developers ;)

(same for `continue`)

Reply:
I'm a one-single-point exit lover, and never ever use break or continue, but in this case I realized that this way was more understandable and maintainable. Even knowing that, I did a deep search about the topic and no clear winner come to me. So I ended up using it, to avoid extra control flags in bucles and ifs. Let's see what the rest of the guys think about it to take an action. WYT?